#14

DIukhymenko opened the pull request:
Close #14

DIukhymenko committed:
Equals and hashCode overwritten for future assertions and correct comparisons of the same objects but with different values.

DIukhymenko committed:
Add equals part to check the correctness of roomChair.equals method

extsoft reviewed (changes requested):
@DIukhymenko you added methods only to `Chair` class. What about others?

extsoft commented on src/main/java/com/automation_boss/inventory/Chair.java:
```diff
@@ -18,4 +19,21 @@ public void print(PrintStream out){
         out.println("Chair shape is - " + shape);
     }
+
+    @Override public boolean equals(Object o) {
```
Please format code in this way
```java
@Override
public boolean equals(Object o) {
...
}
```

This is applicable for all methods with annotations (like `@Override`).

DIukhymenko committed:
Merge branch 'master' into #14

DIukhymenko committed:
Add equals and hashCode methods to the Table class

DIukhymenko committed:
Add equals and hashCode methods to the PaperSheets class
Fix formatting

DIukhymenko committed:
Add equals and hashCode methods to the NoteBook class
Fix formatting

DIukhymenko committed:
Add equals and hashCode methods to the Laptop class

DIukhymenko committed:
Add equals and hashCode methods to the DeskPhone class.
Only weight and color are included to overridden method because other parameters are not interesting for the user who is comparing phones.

DIukhymenko committed:
Add equals and hashCode methods to the CellPhone class;
Fix formatting;
Add access modifiers;
Add CellPhone parameters for more "real" comparison.

extsoft commented on src/main/java/com/automation_boss/inventory/CellPhone.java:
```diff
@@ -34,4 +41,26 @@ public long batteryPercentage() {
+
+    @Override
+    public int hashCode() {
+
```
Please remove the spare line.

extsoft commented on src/main/java/com/automation_boss/inventory/DeskPhone.java:
```diff
@@ -28,4 +33,25 @@ public void contactsList() throws IOException {
+
+    @Override
+    public int hashCode() {
+
```
Please remove  the spare line.

extsoft commented on src/main/java/com/automation_boss/inventory/Laptop.java:
```diff
@@ -1,15 +1,54 @@
+
+    @Override
+    public int hashCode() {
+
```
Please remove the empty line.

extsoft commented on src/main/java/com/automation_boss/inventory/NoteBook.java:
```diff
@@ -21,4 +22,27 @@ public void print(PrintStream out) {
+
+    @Override
+    public int hashCode() {
+
```
Please remove the empty line.

extsoft commented on src/main/java/com/automation_boss/inventory/PaperSheets.java:
```diff
@@ -14,4 +15,22 @@ public PaperSheets(int count, int density) {
+    }
+
+    @Override public int hashCode() {
+
```
Please remove the empty line.

extsoft commented on src/main/java/com/automation_boss/inventory/PaperSheets.java:
```diff
@@ -14,4 +15,22 @@ public PaperSheets(int count, int density) {
         out.println(count + " paper sheets with density = " + density);
     }
+
+    @Override public boolean equals(Object o) {
```
Please format in a way 
```
@Override
public ....
```

extsoft commented on src/main/java/com/automation_boss/inventory/PaperSheets.java:
```diff
@@ -14,4 +15,22 @@ public PaperSheets(int count, int density) {
+        return count == that.count && density == that.density;
+    }
+
+    @Override public int hashCode() {
```
Please format in a way 
```
@Override
public ....
```

extsoft commented on src/main/java/com/automation_boss/inventory/PaperSheets.java:
```diff
@@ -14,4 +15,22 @@ public PaperSheets(int count, int density) {
+        return Objects.hash(count, density);
+    }
+
+    @Override public String toString() {
```
Please format in a way 
```
@Override
public ....
```

extsoft commented on src/main/java/com/automation_boss/inventory/Table.java:
```diff
@@ -15,4 +16,25 @@ public void print(PrintStream out) {
+
+    @Override
+    public int hashCode() {
+
```
Please remove the empty line.

extsoft commented on src/main/java/com/automation_boss/app/App.java:
```diff
@@ -39,5 +45,47 @@ public static void main(String args[]) throws IOException, InterruptedException
         r1.join(secondAttendee);
         r1.describeInventory(myFile);
+        System.out.println("room Chair comparison");
+        if (roomChair.equals(roomChair2))
```
Simplify the code. `roomChair.equals(roomChair2)` will return `true` or `false`. So, you can use it in the message directly.

extsoft commented on src/main/java/com/automation_boss/app/App.java:
```diff
@@ -39,5 +45,47 @@ public static void main(String args[]) throws IOException, InterruptedException
         r1.describeInventory(myFile);
+        System.out.println("room Chair comparison");
+        if (roomChair.equals(roomChair2))
+            System.out.println("true - " + roomChair.toString() + " - " + roomChair2.toString());
```
`.toString()` will be invoked automatically in `System.out.println`. That's why you can simplify the code like `System.out.println("true - " + roomChair + " - " + roomChair2);`.

Please update all lines below.

extsoft reviewed (changes requested):
@DIukhymenko please update the code.

DIukhymenko committed:
Fix formatting;
Implement method for comparing any object which belongs to Inventory

DIukhymenko commented:
@extsoft 
I did it)
Please review

DIukhymenko committed:
Rename comaparisonMethod to inventoryComparing to make method name meanigfull
Add a few method calls with different Inventory

DIukhymenko committed:
formatting

extsoft reviewed (approved):
@DIukhymenko you're the best!
